http-server/src/main/java/com/facebook/airlift/http/server/SimpleAuthorizer.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2010 Proofpoint, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.http.server;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+
+import java.security.Principal;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import static java.util.Objects.requireNonNull;
+
+public class SimpleAuthorizer
+        implements Authorizer
+{
+    private final Pattern userIdentityRegex;
```

Comment:
To be honest it is weird to see `user`/`internal`/`admin` in Airlift. Those are the roles we have defined for Presto. However Airlift is technically a generic framework. 

I would recommend you to change it a little bit.

Let's store the roles mapping in a separate file, let's call it `roles.properties`.

The content of this file can be as following

```
role1=regex_for_role_one
role2=regex_for_role_two
...
```

Then in the SimpleAuthorizer we can specify the path to the `roles.properties` file.